intg/src/main/java/org/apache/atlas/type/AtlasTypeRegistry.java

```diff
@@ -756,13 +757,21 @@ public void removeTypeByName(String name) throws AtlasBaseException {
 
         public List<AtlasBaseTypeDef> getDeleteedTypes() { return deletedTypes; }
 
+        void validateTypeCreation(AtlasBaseTypeDef typeDef) throws AtlasBaseException{
+            if(this.isRegisteredType(typeDef.getName()) && (this.getType(typeDef.getName()).getTypeCategory().equals(TypeCategory.PRIMITIVE)||(this.getType(typeDef.getName()).getTypeCategory().equals(TypeCategory.OBJECT_ID_TYPE)))){
+                throw new AtlasBaseException(AtlasErrorCode.FORBIDDEN_TYPENAME, typeDef.getName());
+            }
+        }
 
         private void addTypeWithNoRefResolve(AtlasBaseTypeDef typeDef) throws AtlasBaseException{
             if (LOG.isDebugEnabled()) {
                 LOG.debug("==> AtlasTypeRegistry.addTypeWithNoRefResolve({})", typeDef);
             }
 
             if (typeDef != null) {
+                if(typeDef.getClass().equals(AtlasEnumDef.class) || typeDef.getClass().equals(AtlasStructDef.class) || typeDef.getClass().equals(AtlasEntityDef.class))
```

Comment:
I have added tests to linear to test out classfiicationDef, relationshipDef, businessmetadataDef as well
in case it fails, we should call validateTypeCreation() for all typeDefs